### PR TITLE
SapMachine #203: A minimal solution to start child processes in an ow…

### LINKED ISSUE #203: A minimal solution to start child processes in an own process group (*nix only)
A new switch is added, -Dsap.jdk.lang.Process.createNewProcessGroupsOnSpawn. If this switch is set to true, new processes started with ProcessBuilder.start or with Runtime.exec() will be leader of an own process group instead of belonging to the process group of the parent VM.

### PR DESCRIPTION
**Pull request by tstuefe**
Implementation for: A minimal solution to start child processes in an own process group (*nix only)

fixes #203

### CODE REVIEW

**Comment by tstuefe** [comment]
ping

**Comment by SapMachine** [comment]
Hello @tstuefe, this pull request fulfills all formal requirements.

**Comment by tstuefe** [comment]
Hi @RealCLanger 

>I understood that Martin wanted to have a change in row 333 of >src/java.base/unix/classes/java/lang/ProcessImpl.java.

Okay. I fix it tomorrow.

> BTW, what about the comment in row 343… can't you remove that one? 😊

Okay.

>Regarding the naming of the property: We could shorten >"sap.jdk.lang.process.createNewProcessGroupOnSpawn" to >"sap.jdk.createNewProcessGroupOnSpawn" maybe that's more pregnant?

Is this really necessary at this stage? That name was mentioned in howtos I sent out, Github issue texts, Jtreg test and example code, which I would have to hunt down now and correct.

If you think this necessary, we can of course change it. But in that case lets please decide for a name and then stick to it. Please lets limit needless review iterations. Every new version I put out needs to be manually built and tested, and I am a bit pressed for time.

I wait for Martin before working on the next version, to add his feedback.

Thanks, Thomas

**Comment by tstuefe** [comment]
Okay, next round...

@RealCLanger: 

>I understood that Martin wanted to have a change in row 333 of >src/java.base/unix/classes/java/lang/ProcessImpl.java.

done

>BTW, what about the comment in row 343… can't you remove that one? blush

done

>Regarding the naming of the property: We could shorten >"sap.jdk.lang.process.createNewProcessGroupOnSpawn" to >"sap.jdk.createNewProcessGroupOnSpawn" maybe that's more pregnant?

The switch was named to conform with the existing launchMechanism switch and has already been mentioned in examples. In fact, I had to rename the switch to its original form (..lang._P_rocess..) to mirror the launchMechanism switch.

Thanks, Thomas

**Comment by SapMachine** [comment]
Hello @tstuefe, this pull request fulfills all formal requirements.

**Comment by RealCLanger** [comment]
Hi Thomas,

> > Regarding the naming of the property: We could shorten >"sap.jdk.lang.process.createNewProcessGroupOnSpawn" to >"sap.jdk.createNewProcessGroupOnSpawn" maybe that's more pregnant?
> 
> The switch was named to conform with the existing launchMechanism switch and has already been mentioned in examples. In fact, I had to rename the switch to its original form (..lang._P_rocess..) to mirror the launchMechanism switch.

just for my understanding... what's the "existing launchMechanism switch" and/or example which hinders us in changing the property name (before we initially create this feature)?

Thanks
Christoph

File: test/jdk/sap/java/lang/CreateNewProcessGroupOnSpawnTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @summary A VM is started with -Dsap.jdk.lang.process.createNewProcessGroupsOnSpawn=true and itself starts
```
Comment (simonis) [inline test/jdk/sap/java/lang/CreateNewProcessGroupOnSpawnTest.java]: You should also change this occurrence of jdk.lang.process to uppercase "P"

Comment (tstuefe) [reply test/jdk/sap/java/lang/CreateNewProcessGroupOnSpawnTest.java]: > You should also change this occurrence of jdk.lang.process to uppercase "P"

Done, see new version.

**Comment by SapMachine** [comment]
Hello @tstuefe, this pull request fulfills all formal requirements.

**Comment by tstuefe** [comment]
> Hi Thomas,
> 
> > > Regarding the naming of the property: We could shorten >"sap.jdk.lang.process.createNewProcessGroupOnSpawn" to >"sap.jdk.createNewProcessGroupOnSpawn" maybe that's more pregnant?
> > 
> > 
> > The switch was named to conform with the existing launchMechanism switch and has already been mentioned in examples. In fact, I had to rename the switch to its original form (..lang._P_rocess..) to mirror the launchMechanism switch.
> 
> just for my understanding... what's the "existing launchMechanism switch"

jdk.lang.Process.launchMechanism

> and/or example which hinders us in changing the property name (before we initially create this feature)?

Example and documentation I prepared for our customers of how to use this feature and which I plan to send out once this thing is in sapmachine11. They are closed source, hence not available.

> 
> Thanks
> Christoph

Thanks, Thomas

**Comment by RealCLanger** [comment]
Hey Thomas,

> jdk.lang.Process.launchMechanism

Ah, I see. Then I fully agree that "sap.jdk.lang.Process.createNewProcessGroupOnSpawn" is a good choice :)

I think the current version is fine to be merged. I'll do it once the test job is done and Martin has set his review to 👍 

Cheers...Christoph

**Comment by tstuefe** [comment]
Thank you.

**Pull request merged**